frontend/src/lib/components/DateFilter/rollingDateRangeFilterLogic.ts

```diff
@@ -54,14 +56,20 @@ export const rollingDateRangeFilterLogic = kea<rollingDateRangeFilterLogicType>(
         counter: [
             counterDefault(props.selected, props.dateFrom) as number | null,
             {
-                increaseCounter: (state) => (state ? state + 1 : 1),
+                increaseCounter: (state) => {
+                    if (state) {
+                        return state < MAX_RETENTION_PERIOD_DAYS ? state + 1 : state
```

Comment:
since this in components and not in session recording specifically we should probably pass in `max_days: number | null` and default to +Infinity or not have the check when null